lib/undercover/result.rb

```diff
@@ -33,7 +33,7 @@ def uncovered?(line_no)
       end
 
       line_cov = coverage.find { |ln, _cov| ln == line_no }
-      line_cov && line_cov[1].zero?
+      line_cov && line_cov.last.zero?
```

Comment:
Ok but if we take only branch coverage it means the CLI will exit with a 0 status when there is uncovered branches and branches will not be in `report.flagged_results`.

I can do two separate function like `uncovered_line?` and `uncovered_branch?` but it's more complicated. We can also just rename the `line_cov` variable but `branch_or_line_cov` like in `coverage_f` : 

https://github.com/grodowski/undercover/blob/4a1ba62afcd18c5826c88754143f43819a6b8515/lib/undercover/result.rb#L43-L56

Another option is to do something like (didn't tried should be refactored) : 

```
    def uncovered?(line)
      lines = {}
      coverage.each do |ln, block_or_line_cov, _, branch_cov|
        next unless ln == line
        lines[ln] = 1 unless lines.key?(ln)
        if branch_cov
          lines[ln] = 0 if branch_cov.zero?
        elsif block_or_line_cov.zero?
          lines[ln] = 0
        end
      end
      return false if lines.keys.size.zero?

      lines.values.sum == lines.keys.size
    end
```

It's similar to what there is in `coverage_f`

Reply:
Thanks @GCorbel and sorry for not being precise enough, what I meant to suggest originally is separating the distinct checks for branch coverage and line coverage which should still works for the case you found. 

Here's a full patch that passes the new spec (`Undercover::Result for a branch without line coverage uncovered gives false` in `result_spec.rb:60`):

```
diff --git a/lib/undercover/result.rb b/lib/undercover/result.rb
index c0ab595..22cba69 100644
--- a/lib/undercover/result.rb
+++ b/lib/undercover/result.rb
@@ -28,12 +28,14 @@ module Undercover
     end
 
     def uncovered?(line_no)
+      # check branch coverage for line_no
       coverage.each do |ln, _block, _branch, cov|
         return true if ln == line_no && cov && cov.zero?
       end
 
-      line_cov = coverage.find { |ln, _cov| ln == line_no }
-      line_cov && line_cov.last.zero?
+      # check line coverage for line_no
+      line_cov = coverage.select { |cov| cov.size == 2 }.find { |ln, _cov| ln == line_no }
+      line_cov && line_cov[1].zero?
     end
 
     # Method `coverage_f` returns the total coverage of this Undercover::Result
```

`uncovered?(27)` is still ` false` because:
- branch coverage is present, so no early return
- `line_cov` for line `27` is blank, so we return `nil`
- 🎉 

I think it is slightly better than checking branch coverage twice with the implicit `last`? LMK what you think and let's ship it soon 🚀

Reply:
I did changes. I misunderstood the first each loop. I tried and it looks correct for me.